### DISCUSSION TITLE
Pydantic error at  `extract_matched_skills`

### DISCUSSION THREAD

**Discussion by hdev-mck** [takline/ResumeGPT#1] [General]
I'm running into this error, at extract_matched_skills. I checked through the chain and when we create `_skills_matcher_chain` we defined inputs but there was not parameters passed to the chain. self.llm_kwargs is always []. 
Let me know how to mitigate this issue. Thanks

Exception has occurred: ValidationError
3 validation errors for PromptInput
ats_keywords
  field required (type=value_error.missing)
non_technical_skills
  field required (type=value_error.missing)
technical_skills
  field required (type=value_error.missing)
  File "ResumeGPT/ResumeGPT/services/resume_improver.py", line 459, in extract_matched_skills
    chain_inputs = format_prompt_inputs_as_strings(prompt_inputs=chain.input_schema().dict(), **self.parsed_job, degrees=self.degrees, experiences=utils.dict_to_yaml_string(dict(Experiences=self.experiences)), education=utils.dict_to_yaml_string(dict(Education=self.education)), skills=self._format_skills_for_prompt(self.skills), objective=self.objective)
  File "ResumeGPT/ResumeGPT/services/resume_improver.py", line 192, in create_draft_tailored_resume
    self.skills = self.extract_matched_skills(verbose=False)
  File "ResumeGPT/main.py", line 7, in <module>
    resume_improver.create_draft_tailored_resume()
pydantic.v1.error_wrappers.ValidationError: 3 validation errors for PromptInput
ats_keywords
  field required (type=value_error.missing)
non_technical_skills
  field required (type=value_error.missing)
technical_skills
  field required (type=value_error.missing)

**Comment by takline** [comment]
I suspect it is a formatting issue with the resume yaml file. I've [made some changes in the recent commit](https://github.com/takline/ResumeGPT/commit/6394f9d89b74dad1a9f7bb8f87f7d8f4a6d66899) that should make this a little easier to work with and understand formatting requirements.

 If that's not the issue, do you mind sending the link?

**Comment by hdev-mck** [comment]
I'm still getting the same error. For consistency, what is the python version? you use.

**Comment by takline** [comment]
3.11.9 

From what I can tell, [resume_improver.download_and_parse_job_post](https://github.com/takline/ResumeGPT/blob/main/services/resume_improver.py#L139-L169) was unable to extract these required fields:

- [ats_keywords](https://github.com/takline/ResumeGPT/blob/main/models/job_post.py#L32-L34)
- [non_technical_skills](https://github.com/takline/ResumeGPT/blob/main/models/job_post.py#L45-L47)
- [technical_skills](https://github.com/takline/ResumeGPT/blob/main/models/job_post.py#L42-L44)

 [on this line](https://github.com/takline/ResumeGPT/blob/main/services/resume_improver.py#L526). Take a look at the output [job.yaml ](https://github.com/takline/ResumeGPT/blob/main/services/resume_improver.py#L168) file to see if the extractor did in fact fail to extract these fields.

**Comment by hdev-mck** [comment]
job.yaml is extracting that information.

I think this is a langchain problem. Issue is when we have a chain and try to get input_schema of the chain, it is throwing pydantic errors. I used a test to replicate this issue. Not sure if I'm missing something. 

![Screenshot 2024-07-19 at 11 24 58 AM](https://github.com/user-attachments/assets/2fc5f509-f006-4938-b0d0-8af8416716e2)

**Comment by hdev-mck** [comment]
After a bit of grinding, there seems to be pydantic issues with langchain. Some dependency cluster f*. I downgraded to a working version from my previous project and it works.

Below is my requirements.txt file:

`#python 3.11.9

beautifulsoup4
configparser
langchain==0.1.20
langchain-openai==0.1.6
langchain-core
pydantic==2.7.1
reportlab
requests
ruamel.yaml
pytest
free-proxy
langchain-community==0.0.38
python-dateutil
Jinja2
notebook`

**Reply by TaherPatan** [reply]
It is giving me the same error even with these versions installed in a completely new environment. It is a massive issue.

**Comment by takline** [comment]
Ah wow appreciate the pointer - I'll make sure to update the requirements to prevent this from happening again

**Reply by TaherPatan** [reply]
It is giving me the same error even with these versions installed in a completely new environment. It is a massive issue. Did you manage to find any solutions for it apart from installing the packages with the same version?

**Comment by virajnp** [comment]
Does this tool work and where should the starting point be to run it.

**Reply by Nipun-Gupta27** [reply]
I don't see any main.py. do we need to create one?
